api/oci/ociutils/ref.go

```diff
@@ -0,0 +1,83 @@
+package ociutils
+
+import (
+	"strings"
+
+	"github.com/mandelsoft/goutils/generics"
+	"github.com/opencontainers/go-digest"
+)
+
+// ParseVersion parses the version part of an OCI reference consisting
+// of an optional tag and/or digest.
+func ParseVersion(vers string) (ArtVersion, error) {
```

Comment:
Because all methods on ArtVersion are defined on a pointer I believe this should become 
```suggestion
func ParseVersion(vers string) (*ArtVersion, error) {
```

Reply:
It the sibling for oci.ParseArt. It returned a value, therefore ParseVersion was done the same way. Unfortunately it has to be moved to another package to avoid cycles,

Reply:
Since ParseArt is also a ref creator, could we agree to move ParseArt to Pointer too? I assume you are talking about https://github.com/open-component-model/ocm/blob/69ea53a86ed3ae39e815f9bc73014bf49f3c731e/api/oci/ref.go#L227 and it seems like a pointer is applicable here too.